ci-operator/step-registry/cucushift/hypershift-extended/capi/upgrade-cp/cucushift-hypershift-extended-capi-upgrade-cp-ref.yaml

```diff
@@ -0,0 +1,29 @@
+ref:
+  as: cucushift-hypershift-extended-capi-upgrade-cp
+  from: rosa-aws-cli
+  grace_period: 5m
+  cli: latest
+  timeout: 2h0m0s
+  commands: cucushift-hypershift-extended-capi-upgrade-cp-commands.sh
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+  env:
+    - name: CLUSTER_TIMEOUT
+      default: "3600"
+      documentation: Set to number of seconds for the cluster to timeout if it's not ready.
+    - name: OCM_LOGIN_ENV
+      default: "staging"
+      documentation: The environment for rosa login. The supported values are [production, staging, integration].
+    - name: CLUSTER_NAME
+      default: ""
+      documentation: The name of the rosa cluster to create. Must be unique for the account, lowercase, and no more than 54 characters.
+    - name: REGION
+      default: "us-east-1"
+      documentation: "The AWS region of the cluster."
+    - name: OPENSHIFT_VERSION
```

Comment:
I advise using `UPGRADED_TO_VERSION,` `OPENSHIFT_VERSION` as a variable used for provision.

Reply:
OPENSHIFT_VERSION is the variable used for the provision. The target version is the latest one according to the value of OPENSHIFT_VERSION.
e.g. OPENSHIFT_VERSION = 4.16,  AVAILABLE_UPGRADE=true in the provision step, here in the upgrade step, OPENSHIFT_VERSION is always from the prev step, so I can find the target version automatically based on this value.

But you are right, it is a little confusing to define it here. Maybe I can save the value OPENSHIFT_VERSION in provision step and then in this step I don't need to define it, and just read it from shared_dir. Is it a better idea? Thanks

Reply:
> But you are right, it is a little confusing to define it here. Maybe I can save the value OPENSHIFT_VERSION in provision step and then in this step I don't need to define it, and just read it from shared_dir. Is it a better idea? 

How about getting the open shift version by `rosa describe cluster`?

Reply:
Thanks! I tried different ways to realize it. Finally, I think it is better to follow your first comment to define the new var UPGRADED_TO_VERSION to control the target version.